Kind30398Event/mod.ts

```diff
@@ -0,0 +1,73 @@
+import { z } from "zod";
+
+import { version as PACKAGE_VERSION } from "./deno.json" with { type: "json" };
+export const CONTENT_MINIMUM_LENGTH = 3;
+export const CONTENT_MAXIMUM_LENGTH = 300;
+
+function isHex(s: string) {
+  return s.split("").every((c) => "0123456789abcdef".split("").includes(c));
+}
+
+function isPlusCode(code: string) {
+  const re =
+    /(^|\s)([23456789C][23456789CFGHJMPQRV][23456789CFGHJMPQRVWX]{6}\+[23456789CFGHJMPQRVWX]{2,7})(\s|$)/i;
+  return re.test(code);
+}
+
+export const eventSchema = z
+  .object({
+    id: z.string().length(32),
+    pubkey: z.string().length(32),
+    kind: z.number(),
+    created_at: z.number(),
+    tags: z.string().array().array(),
+    content: z.string(),
+    sig: z.string(),
+  })
+  .strict();
+
+export type Event = z.infer<typeof eventSchema>;
+
+function hasOLC(tags: string[][]): boolean {
```

Comment:
Love the function, would prefer a proper name rather than a Stallman impersonation...

hasOpenLocationCodeTag?